### PR TITLE
Cleanup key locations

### PR DESCRIPTION
**Pull request by estenberg**
Must be merged with https://github.com/mendersoftware/mender-api-gateway-docker/pull/52.

### CODE REVIEW

**Comment by bboozzoo** [comment]
Personally, not sure if `/keys` is a place where I would look for certificates. If others are ok with this change then we can merge it

**Comment by estenberg** [comment]
ok, the reason is that I felt that all these key/cert files were related; you probably want to regenerate all of them in one go and you would need both a private key and a cert file for api gateway and storage proxy.

I couldn't find a good name except keys as a parent dir, and would be open to proposals!

**Comment by bboozzoo** [comment]
IMO certificates should remain in some certificate related directory, `cert`, `certs`, `certificates`, `ssl` etc. `keys` would not be my first choice to look in if I were trying to locate certificates.

**Comment by estenberg** [comment]
Do you have some reference from other / standard locations for similar tools?
Perhaps you could see if others in the team has an opinion or create an alternate proposal?

**Comment by bboozzoo** [comment]
As I wrote earlier, if others are ok then we can merge this PR.

@mendersoftware/rndity @maciejmrowiec @pasinskim @kacf

**Comment by kacf** [comment]
I have no strong preference, but I'm leaning towards using the parent dir `certs` instead of `keys` as well.

**Comment by estenberg** [comment]
Alright.. The logical reason was that a cert embeds a key, but a key is not a cert.
So if we want a single dir, then it should probably be called keys (or something generic), not certs.
The main purpose of a single dir is that people would probably regenerate all these (not just the certs, for example).

But what about this structure:

certs/
├── api-gateway
│   ├── cert.crt
│   └── private.key
└── storage-proxy
    ├── cert.crt
    └── private.key

keys/
├── deviceauth
│   └── private.key
└── useradm
    └── private.key

**Comment by estenberg** [comment]
@bboozzoo I updated the PR with new locations.
If this is ok and merged I will update keygen script & docs accordingly.

File: docker-compose.demo.yml [hunk 1]
```diff
@@ -23,8 +23,8 @@ services:
         environment:
             MAPPED_PORT: 8080
         volumes:
-            - ./ssl/mender-api-gateway/cert.pem:/var/www/mendersoftware/cert/cert.pem
-            - ./ssl/mender-api-gateway/key.pem:/var/www/mendersoftware/cert/key.pem
+            - ./certs/api-gateway/cert.crt:/var/www/mendersoftware/cert/cert.crt
+            - ./certs/api-gateway/private.key:/var/www/mendersoftware/cert/private.key
```
Comment (estenberg) [inline docker-compose.demo.yml]: @bboozzoo I think this should be ready to merge with https://github.com/mendersoftware/mender-api-gateway-docker/pull/52/files already; they should be mounted to the right place, no?

Comment (bboozzoo) [inline docker-compose.demo.yml]: yup, we shoould merge both then

**Pull request merged**